### PR TITLE
Next generation CLI (2.0.0)

### PR DESCRIPTION
**Pull request by frenck**
This is a tracking PR for the Next Generation CLI (v2.0.0).
If you want to contribute to this version of the CLI, please be sure to target the `next` branch.

In case of questions, feel free to address me or @mjrider.

TODO list:
- homeassistant
  - [x] info
  - [x] logs
  - [x] check
  - [x] restart
  - [x] start
  - [x] stop
  - [x] update
  - [x] options

- supervisor
  - [x] info
  - [x] logs
  - [x] reload
  - [x] update
  - [x] options
  
- host   
  - [x] info
  - [x] reboot
  - [x] shutdown
     
- hassos    
  - [x] info
  - [X] -update-

- hardware
  - [x] info
  - [x] audio

- snapshots, sn      
  - [x] list
  - [x] info
  - [x] reload
  - [x] new
  - [x] restore
  - [x] remove

- addons       
  - [x] list
  - [x] info
  - [x] logo
  - [x] changelog
  - [x] logs,
  - [x] stats
  - [x] reload
  - [x] start
  - [x] stop
  - [x] install
  - [x] uninstal
  - [x] update

### CODE REVIEW

**Comment by bbriggs** [comment]
So, a bit of explanation on this PR and why later changed direction with it:

Treating each subcommand as its own new instance of a CLI makes sense for organizing nested subcommands that have the same names (such as `info`, which appears under several subcommands).

However, it makes for a slightly more complicated pattern of adding new subcommands and does not allow Cobra to properly pass global options down to subcommands, since they are a new instance.

I have a couple of ideas of how to make this work cleanly, but in the last iteration (see the recently closed PR on it) I put all commands and subcommands in one directory and just renamed the commands appropriately to avoid naming conflicts.

**Comment by mjrider** [comment]
example on how i think to move the commands
https://github.com/mjrider/hassio-cli/commit/ab94199c7e1ac06e38debf9e56a5e8d364bef87c

**Comment by bbriggs** [comment]
@mjrider agreed, it's a better pattern. subpackages overcomplicate the CLI. We would just need to be mindful of our use of globals when passing options to subcommands.

**Comment by mjrider** [comment]
TODO list:
- homeassistant
  - [x] info
  - [x] logs
  - [x] check
  - [x] restart
  - [x] start
  - [x] stop
  - [x] update

- supervisor
  - [ ] info
  - [ ] logs
  - [ ] reload
  - [ ] update
  
- host   
  - [ ] reboot
  - [ ] shutdown
     
- hassos    
  - [ ] info
  - [ ] update

- hardware
  - [ ] info
  - [ ] audio

- snapshots, sn      
  - [ ] list
  - [ ] info
  - [ ] reload
  - [ ] new
  - [ ] restore
  - [ ] remove

- addons       
  - [ ] list
  - [ ] info
  - [ ] logo
  - [ ] changelog
  - [ ] logs,
  - [ ] stats
  - [ ] reload
  - [ ] start
  - [ ] stop
  - [ ] install
  - [ ] uninstal
  - [ ] update

**Comment by mjrider** [comment]
hassos update is non-existing in the current dev branch
hassos info is just an alias to host info
https://github.com/home-assistant/hassio-cli/blob/8582e39f8de622267c73d0bcdb7712a07a217d53/command/hassos.go#L27

File: cmd/homeassistant_options.go [hunk 1]
```diff
@@ -43,14 +43,19 @@ var homeassistantOptionsCmd = &cobra.Command{
 			options["startup_time"] = startupTime
 		}
 
+		waitBoot, err := cmd.Flags().GetInt("wait_boot")
```
Comment (pvizeli) [inline cmd/homeassistant_options.go]: That is a supervisor option

Comment (mjrider) [reply cmd/homeassistant_options.go]: as agreed in the chat, this is the correct option,

**Pull request merged**